app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionFragment.kt

```diff
@@ -0,0 +1,357 @@
+                            .putExtra(KEY_FILE_PATH, exportFile.absolutePath))
+                }
+            } else if (requestCode == REQUEST_IMPORT_CODE) {
+                val path = Utils.getFileForUri(data.data!!).absolutePath
+                ImportConfirmationDialog.show(this, Intent(activity, SubscriptionsImportService::class.java)
+                        .putExtra(KEY_MODE, PREVIOUS_EXPORT_MODE)
+                        .putExtra(KEY_VALUE, path))
+            }
+        }
+    }
+
+    //////////////////////////////////////////////////////////////////////////
+    // Fragment Views
+    //////////////////////////////////////////////////////////////////////////
+
+    private fun setupInitialLayout() {
+        Section().apply {
+            val carouselAdapter = GroupAdapter<GroupieViewHolder>()
+
+            carouselAdapter.add(FeedGroupCardItem(-1, getString(R.string.all), FeedGroupIcon.RSS))
+            carouselAdapter.add(feedGroupsSection)
+            carouselAdapter.add(FeedGroupAddItem())
+
+            carouselAdapter.setOnItemClickListener { item, _ ->
+                listenerFeedGroups.selected(item)
+            }
+            carouselAdapter.setOnItemLongClickListener { item, _ ->
+                if (item is FeedGroupCardItem) {
+                    if (item.groupId == -1L) {
+                        return@setOnItemLongClickListener false
+                    }
+                }
+                listenerFeedGroups.held(item)
+                return@setOnItemLongClickListener true
+            }
+
+            feedGroupsCarousel = FeedGroupCarouselItem(requireContext(), carouselAdapter)
+            add(Section(HeaderItem(getString(R.string.feed_groups_header_title)), listOf(feedGroupsCarousel)))
+
+            groupAdapter.add(this)
+        }
+
+        subscriptionsSection.setPlaceholder(EmptyPlaceholderItem())
+        subscriptionsSection.setHideWhenEmpty(true)
+
+        importExportItem = FeedImportExportItem(
+                { onImportPreviousSelected() },
+                { onImportFromServiceSelected(it) },
+                { onExportSelected() })
+        groupAdapter.add(Section(importExportItem, listOf(subscriptionsSection)))
+
+    }
+
+    override fun initViews(rootView: View, savedInstanceState: Bundle?) {
+        super.initViews(rootView, savedInstanceState)
+
+        items_list.layoutManager = LinearLayoutManager(requireContext())
+        items_list.adapter = groupAdapter
+
+        viewModel = ViewModelProviders.of(this).get(SubscriptionViewModel::class.java)
+        viewModel.stateLiveData.observe(viewLifecycleOwner, androidx.lifecycle.Observer { it?.let(this::handleResult) })
+        viewModel.feedGroupsLiveData.observe(viewLifecycleOwner, androidx.lifecycle.Observer { it?.let(this::handleFeedGroups) })
+    }
+
+    private fun showLongTapDialog(selectedItem: ChannelInfoItem) {
+        val context = context
```

Comment:
Right, if you control click, you'll be on getContext().
IDE propose to change to context to « Cleanup code »

Reply:
The Android Studio converter was used, and I missed this one change.

This bizarre function have some oddities even before it was converted, for example, it was using the context resources instead of using the direct `getString` that is available in the fragment.

There's still some problems like the custom dialog layout that do not follow the android style, but I've changed some things to kotlin style now.